src/SvgParser.elm

```diff
@@ -285,18 +312,23 @@ parseToNode input =
 parse : String -> Result String (Html msg)
 parse input =
     let
-        toHtml svgNode =
-            case svgNode of
-                SvgElement element ->
-                    if element.name == "svg" then
-                        Ok <|
-                            svg (List.map toAttribute element.attributes)
-                                (List.map nodeToSvg element.children)
-
-                    else
-                        Err "Top element is not svg"
-
-                _ ->
+        toHtml svgNodes =
+            case svgNodes of
+                svgNode :: tl ->
+                    case svgNode of
```

Comment:
You can merge the two `case ... of ` statements together and handle the cases in one `case ... of` statement. Like so:

```elm
case svgNodes of
    SvgElement element :: tl -> ...
    _ :: tl -> ...
    [] -> ...